unlock-app/src/middlewares/wedlocksMiddleware.ts

```diff
@@ -16,9 +16,7 @@ const wedlocksMiddleware = (config: any) => {
           const { origin } = window.location
           // TODO: then and catch? I think we really only need to worry about errors.
           wedlockService.confirmEmail(action.emailAddress, `${origin}/signup`)
-        }
-
-        if (action.type === WELCOME_EMAIL && window && window.location) {
+        } else if (action.type === WELCOME_EMAIL && window && window.location) {
```

Comment:
I know this is a line shift but we should consider moving the evaluation criteria  inside a meaningfully name method